src/System.Globalization.Extensions/tests/Normalization/StringNormalizationTests.cs

```diff
@@ -24,21 +24,50 @@ public void IsNormalized(string value, NormalizationForm normalizationForm, bool
         }
 
         [Fact]
-        public void IsNormalized_Invalid()
+        [PlatformSpecific(~TestPlatforms.Windows)]
+        public void IsNormalized_Invalid_NonWindows()
         {
             Assert.Throws<ArgumentException>(() => "\uFB01".IsNormalized((NormalizationForm)10));
 
             Assert.Throws<ArgumentException>("strInput", () => "\uFFFE".IsNormalized()); // Invalid codepoint
             Assert.Throws<ArgumentException>("strInput", () => "\uD800\uD800".IsNormalized()); // Invalid surrogate pair
 
             Assert.Throws<ArgumentNullException>("strInput", () => StringNormalizationExtensions.IsNormalized(null));
-            
-            Exception exception = Record.Exception(() => ((string)null).IsNormalized());
-            
+
+            Assert.Throws<ArgumentNullException>("strInput", () => ((string)null).IsNormalized());
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework | TargetFrameworkMonikers.Netcoreapp)]
+        public void IsNormalized_Invalid_NetCore()
+        {
+            Assert.Throws<ArgumentException>(() => "\uFB01".IsNormalized((NormalizationForm)10));
+
+            Assert.Throws<ArgumentException>("strInput", () => "\uFFFE".IsNormalized()); // Invalid codepoint
+            Assert.Throws<ArgumentException>("strInput", () => "\uD800\uD800".IsNormalized()); // Invalid surrogate pair
+
+            Assert.Throws<ArgumentNullException>("strInput", () => StringNormalizationExtensions.IsNormalized(null));
+
+            Assert.Throws<ArgumentNullException>("strInput", () => ((string)null).IsNormalized());
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(~(TargetFrameworkMonikers.NetFramework | TargetFrameworkMonikers.Netcoreapp))]
+        public void IsNormalized_Invalid_Desktop()
+        {
+            Assert.Throws<ArgumentException>(() => "\uFB01".IsNormalized((NormalizationForm)10));
+
+            Assert.Throws<ArgumentException>("strInput", () => "\uFFFE".IsNormalized()); // Invalid codepoint
+            Assert.Throws<ArgumentException>("strInput", () => "\uD800\uD800".IsNormalized()); // Invalid surrogate pair
+
+            Assert.Throws<ArgumentNullException>("strInput", () => StringNormalizationExtensions.IsNormalized(null));
+
             // On desktop IsNormalized is not extension method, trying to do ((string)null).IsNormalized()
             // will get NullReferenceException, in .Net Core we use extension method which will throw
             // ArgumentNullException
-            Assert.True((exception is ArgumentNullException) || (exception is NullReferenceException));
```

Comment:
we cannot remove the extensions methods because of the app comparability. please keep it as it is. thanks.